src/navigation/MainTabNavigator.tsx

```diff
@@ -1,37 +1,154 @@
 import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 
-import { colors, device, normalize } from '../config';
-import { TabNavigatorConfig } from '../types';
+import { LoadingSpinner } from '../components';
+import { colors, normalize } from '../config';
+import {
+  createDynamicTabConfig,
+  tabNavigatorConfig as defaultTabNavigatorConfig
+} from '../config/navigation/tabConfig';
+import { useStaticContent } from '../hooks';
+import { ScreenName, TabConfig, TabNavigatorConfig } from '../types';
 
 import { getStackNavigator } from './AppStackNavigator';
 
+const defaultTabRoutes: TabRoutes = {
+  tabConfig: {
+    activeBackgroundColor: colors.surface,
+    activeTintColor: colors.accent,
+    inactiveBackgroundColor: colors.surface,
+    inactiveTintColor: colors.primary,
+    tabs: ['Home', 'Service', 'Events', 'Index', 'About']
+  }
+};
+
+type Tab = {
+  accessibilityLabel: string;
+  iconName: string;
+  iconSize: number;
+  label: string;
+  screen: ScreenName;
+  params?: Record<string, any>;
+};
+
+type TabRoutes = {
+  tabConfig: {
+    activeTintColor: string;
+    inactiveTintColor: string;
+    activeBackgroundColor: string;
+    inactiveBackgroundColor: string;
+    tabs: (Tab | string)[];
+  };
+};
+
+const useTabRoutes = () => {
+  const { data, error, loading } = useStaticContent<TabRoutes>({
+    name: 'tabNavigation',
+    type: 'json'
+  });
+
+  const [tabRoutes, setTabRoutes] = useState<TabRoutes>(defaultTabRoutes);
+
+  useEffect(() => {
+    const tabs: (Tab | string)[] = data?.tabConfig.tabs || defaultTabRoutes.tabConfig.tabs;
+    const { activeBackgroundColor, activeTintColor, inactiveBackgroundColor, inactiveTintColor } =
+      data?.tabConfig || defaultTabRoutes.tabConfig;
```

Comment:
why not grabbing `tabs` here as well but separately a line before?

Reply:
Yes, that's right. I'll do it.